### PR TITLE
Support for specialized functions

### LINKED ISSUE #7: Module prefixed names
I gave this a quick spin and ran into a difficulty.  Namely, our code base generally imports Ramda as the `R` module and all references to Ramda functions are then prefixed with the module name.  Not seeing any lint issues, I changed a use of `R.pluck(...)` to `R.map(R.prop(...))` and I still didn't see any lint issues.  Then I changed it to `map(prop(...))` and got the lint issue I expected.  Is there a way to configure this plugin with a module prefix?

### PR DESCRIPTION
**Pull request by haskellcamargo**
This pull-request implements support for specialized calls via `R`. Soon, a good improvement will be making the a parameter, but it solves the problem for now. I suggest using `2.1.0` as version name for this.

Now, both `R.map(R.prop)` and `map(prop)` can be recognized and works mutually. Adapting to check the scope later also became easier; I've added an `ast-helper` to get the work done. We currently have `135` tests.

This introduces no breaking changes.
This introduces no new rules.

Closes #7.
Thanks @polytypic for reporting and @lo1tuma for the idea.

### CODE REVIEW

File: ast-helper.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+// :: Node -> String
+const getName = R.ifElse(
+    R.propEq('type', 'MemberExpression'),
+    R.path(['property', 'name']),
+    R.prop('name')
+);
+
+// :: String -> Node -> Boolean
+const isName = name => R.either(
```
Comment (lo1tuma) [inline ast-helper.js]: I think `isName` is not very expressive, I would suggest something like `isRamdaMethod`.

File: rules/any-pass-simplification.js [hunk 2]
```diff
@@ -1,24 +1,24 @@
...
-                    node,
-                    message: '`anyPass` containing only complement functions should be a complement of `allPass`'
-                });
-            }
+        const match = isCalling({
+            name: 'anyPass',
+            arguments: R.both(
+                R.propSatisfies(R.lt(0), 'length'),
+                R.where({
+                    0: R.propSatisfies(R.all(isCalling({ name: 'complement' })), 'elements')
```
Comment (lo1tuma) [inline rules/any-pass-simplification.js]: What happens when the first argument is not an `ArrayExpression`, e.g. `anyPass(myPredicates, myData)`? Looks like the rule would crash in this case.

File: rules/any-pass-simplification.js [hunk 3]
```diff
@@ -1,24 +1,24 @@
 'use strict';
+const R = require('ramda');
+const isCalling = require('../ast-helper').isCalling;
```
Comment (lo1tuma) [inline rules/any-pass-simplification.js]: This file needs to be added to the whitelist of published files in `package.json`.

File: rules/cond-simplification.js [hunk 4]
```diff
@@ -1,12 +1,20 @@
...
-            && node.callee.name === 'cond'
-            && node.arguments.length > 0
-            && node.arguments[0].type === 'ArrayExpression'
-            && node.arguments[0].elements.length <= 2) {
+        const match = isCalling({
+            name: 'cond',
+            arguments: R.both(
+                R.propSatisfies(R.lt(0), 'length'),
+                R.where({
+                    0: R.pathSatisfies(R.gte(2), ['elements', 'length'])
```
Comment (lo1tuma) [inline rules/cond-simplification.js]: Same here, the first argument might not be an `ArrayExpression`.

File: rules/any-pass-simplification.js (hunk 3 above)
Comment (haskellcamargo) [reply rules/any-pass-simplification.js]: Fixed.

**Comment by haskellcamargo** [comment]
Fixed everything listed!

I couldn't think of a better name than `isCalling`, so I kept it for now.
I've implemented support for literal array access, such as `R['filter']` too.
The line and columns are reported by `node`, which is passed via the parent call, so, for now, it will always link to the call expression correctly, but we can further add test cases for that.

**Pull request merged**